gadgets/trace_dns/program.go

```diff
@@ -1,9 +1,203 @@
 package main
 
 import (
+	"fmt"
+
 	wapc "github.com/wapc/wapc-guest-tinygo"
 )
 
 func main() {
-	wapc.RegisterFunctions(wapc.Functions{})
+	wapc.RegisterFunctions(wapc.Functions{
+		"Init":            Init,
+		"column_name":     column_name,
+		"column_qr":       column_qr,
+		"column_qtype":    column_qtype,
+		"column_rcode":    column_rcode,
+		"column_pkt_type": column_pkt_type,
+	})
+}
+
+func Init(payload []byte) ([]byte, error) {
+	//wapc.ConsoleLog("Hello from Wasm!")
```

Comment:
I removed the comment but I kept the empty function: it seems that the host is not able to inspect the wasm module to know if the function exists at all. If it does not exist, it returns an error, but unless we do string parsing, we don't know if Init() returned an error or if it does not exist.

So I am suggesting to make the Init() function mandatory. The difference with the `main()` function is that it can receives custom parameters, which can be useful for some gadgets later on.